src/extension/executors/runner.ts

```diff
@@ -146,8 +144,21 @@ export async function executeRunner(
           }
         }, OutputType.outputItems)
       }
-
-      replaceOutput(exec, [new NotebookCellOutput([item])])
+      const enableNotebookTerminal = mime === 'text/plain' && isNotebookTerminalFeatureEnabled('nonInteractive')
+      if (!enableNotebookTerminal) {
+        replaceOutput(exec, [new NotebookCellOutput([item])])
+      } else {
+        const json = NotebookTerminal.create(cellUUID, {
+          content: Buffer.concat(output).toString('base64'),
+          removeState: true
+        })
+
+        await replaceOutput(exec, [
```

Comment:
By executing this as part of `handleOutput`, it will get called every time the output updates. 

I don't believe that is necessary. If you look above this code, you will see that the events for stdout get sent regardless of interactive status. 

Instead, I think you should early return from `handleOutput` if we're using the notebook terminal, and open the notebook terminal in this `if` branch, similar to the interactive case.

Even more ideally, you could open the terminal above the `if` branch.